plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,285 @@
+            .addSettingsUpdateConsumer(ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE, it -> enableAutoReLoadModel = it);
+    }
+
+    /**
+     * the main method: model auto reloading
+     */
+    public void autoReLoadModel() {
+        log.debug("enableAutoReLoadModel {} ", enableAutoReLoadModel);
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+            String localNodeId = clusterService.localNode().getId();
+            DiscoveryNode node = nodeHelper.getNode(localNodeId);
+            if (!MLNodeUtils.isMLNode(node)) {
+                return;
+            }
+
+            // According to the node id to query the number of retries, if more than 2 (the maximum number of retries), we don't need to
+            // retry,
+            // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+            Integer reTryTimes = 0;
+            if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+                reTryTimes = getReTryTimes(localNodeId);
+            }
+            if (reTryTimes > ML_MODEL_RELOAD_MAX_RETRY_TIMES) {
+                log.debug("have exceeded max retry times, always failure");
+            }
+
+            // auto reload all models of this local node, if it fails, reTryTimes+1, if it succeeds, reTryTimes is cleared to 0
+            try {
+                CompletableFuture<Void> completableFuture = CompletableFuture.runAsync(() -> autoReLoadModelByNodeId(localNodeId));
+                completableFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Can't auto reload model, the reason is:", e);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Can't auto reload model");
+        }
+    }
+
+    /**
+     * auto reload all the models under the node id
+     * the node must be a ml node
+     * @param nodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        if (!isExistedIndex(ML_TASK_INDEX)) {
+            return;
+        }
+
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet();
+
+        if (response == null) {
+            return;
+        }
+        SearchHits searchHits = response.getHits();
+        if (searchHits == null) {
+            return;
+        }
+        SearchHit[] hits = searchHits.getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        Integer reTryTimes = 0;
+        for (SearchHit hit : hits) {
```

Comment:
We shouldn't retry for every task, we should only retry the latest success one.

Reply:
in the iteration, I add the condition
` if (workerNode.contains(nodeId) && mlTaskType == MLTaskType.LOAD_MODEL && mlTaskState == MLTaskState.COMPLETED) {
                    autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
                }`
if the condition didn't be meet, it will not do retry operation.

Reply:
Even if with this check, we shouldn't auto reload for every document that meets the criteria, think of user did two deployment, and the first time deployment is not the correct status anymore, we should always focus on the latest successful deployment.

Reply:
I am start to understand your means. I just to get 1 doc from index `ML_TASK_INDEX`, is right?